app/controllers/posts_controller.rb

```diff
@@ -36,22 +37,35 @@ def update
     @post.text = new_post.text
 
     if @post.save
-      flash[:success] = 'Post has been updated'
-      redirect_to user_post_url(id: @post.id)
+      success
+      redirect_to user_post_url(@post)
     else
-      @post = saved_post
-      flash.now[:alert] = 'Invalid Input'
-      render :edit
+      show_errors
+      render :edit, status: 500
     end
   end
 
   private
 
+  def success
+    flash[:notice] = 'Post created Successfully'
+  end
+
+  def failed
+    flash.now[:alert] = 'Sorry, the post was not saved'
+  end
+
+  def show_errors
+    failed
+    errors = @post.errors.map(&:full_message)
+    flash.now[:error] = errors.join(' | ')
+  end
+
```

Comment:
- [ ] Kindly fix the flash messages when  creating a post

![image](https://user-images.githubusercontent.com/38635158/167236586-f9945924-3074-481d-9b55-15b91085d6f7.png)